.gitmodules

```diff
@@ -16,3 +16,15 @@
 [submodule "depends/jsoncpp"]
 	path = depends/jsoncpp-sub
 	url = ../../DFHack/jsoncpp.git
+[submodule "depends/xlsxio"]
+	path = depends/xlsxio
+	url = ../../brechtsanders/xlsxio.git
+	shallow = true
```

Comment:
> 1b) I plan to fix the dependencies to a particular branch/commit instead of having them track master/HEAD. I don't want them changing out from underneath us. Is this ok?

Technically, that's what submodules always do. For example, a fresh clone of DFHack 0.47.04-r2 (commit https://github.com/DFHack/dfhack/commit/f2b0f012c9895aa3fd743ad50fab9e6fb78b479d) will check out https://github.com/dfhack/scripts/commit/568c586 in the scripts folder. You can check out the `master` branch or the `0.47.04-r2` tag, but since it points to the same commit, Git won't recognize a change in the DFHack repo. It only keeps track of submodules' commit hashes in the parent project, so they'll always be locked to whatever we specify (at least in an unmodified clone) without doing anything special.

The only different thing I see here (compared to other submodules) is `shallow = true`. That wouldn't change the behavior of these submodules, other than making the downloaded history smaller (it's equivalent to `--depth 1` when cloning), which looks like it's a good thing for at least expat, which has a large history. I assume that's why you added this. A shallow clone of submodules *might* make updating to a new version trickier, since only a single commit would have been fetched originally - I would likely have to experiment to understand how that works.

Reply:
ah, ok. I didn't know they were locked to a specific commit by design. chalk that misunderstanding up to my inexperience with creating git submodules. I suggest, then, that the policy be that we never update these dependencies unless we actually need to.

Yes, I added the shallow=true lines since I figured we don't really care about the commit histories, and it reduces the download size. I don't seem to have any trouble switching branches or pulling updates in those `--depth 1` repos.